library/src/main/java/com/novoda/downloadmanager/lib/BatchRetrievalException.java

```diff
@@ -0,0 +1,15 @@
+package com.novoda.downloadmanager.lib;
+
+final class BatchRetrievalException extends RuntimeException {
+    static BatchRetrievalException failedQueryForBatches() {
+        return new BatchRetrievalException("Failed to query for batches");
+    }
+
+    static BatchRetrievalException failedQueryForBatch(long batchId) {
```

Comment:
Because it doesn't give it a name mainly. Also, with constructors, usually people tend to end up with the message as a parameter, leaking that information outside the class and hiding when the exception has come to a point where it makes sense to split it.

People are used to it because it's easier, imho not a good reason to keep the habit :)